### ISSUE TITLE
[JENKINS-36615] Unable to run multibranch pipeline with '/' the name

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#2998]
<p>If you setup a multibranch github project based on: </p>

<p><a href="https://github.com/cloudbeers/blueocean-demo-project" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/cloudbeers/blueocean-demo-project</a></p>

<p>And try to run the branch called "feature/newhotness" you get: </p>

```

jenkins-js-extension.js:10841 POST http://localhost:8080/jenkins/blue/rest/organizations/jenkins/pipelines/multi2/branches/feature%2Fnewhotness/runs/ 404 (Not Found)run @ jenkins-js-extension.js:10841run @ jenkins-js-extension.js:13760onClick @ jenkins-js-extension.js:13824ReactErrorUtils.invokeGuardedCallback @ blueocean.js:28058executeDispatch @ blueocean.js:20044executeDispatchesInOrder @ blueocean.js:20064executeDispatchesAndRelease @ blueocean.js:19517executeDispatchesAndReleaseTopLevel @ blueocean.js:19528forEachAccumulated @ blueocean.js:34385processEventQueue @ blueocean.js:19690runEventQueueInBatch @ blueocean.js:28084handleTopLevel @ blueocean.js:28095handleTopLevelImpl @ blueocean.js:28173perform @ blueocean.js:33838batchedUpdates @ blueocean.js:27265batchedUpdates @ blueocean.js:31544dispatchEvent @ blueocean.js:28250
jenkins-js-extension.js:13767 Unexpected error queuing a run of "feature%2Fnewhotness". Response:(anonymous function) @ jenkins-js-extension.js:13767(anonymous function) @ jenkins-js-extension.js:10851
jenkins-js-extension.js:13768 Response {type: "basic", url: "http://localhost:8080/jenkins/blue/rest/organizati…elines/multi2/branches/feature%2Fnewhotness/runs/", status: 404, ok: false, statusText: "Not Found"…}


```

<p>This is possibly related to: <a href="https://issues.jenkins-ci.org/browse/JENKINS-36613" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-36613</a></p>

<p>to reproduce: </p>

<p>1) setup a multibranch github based on <a href="https://github.com/cloudbeers/blueocean-demo-project" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/cloudbeers/blueocean-demo-project</a><br/>
2) let it run, and then try to rerun the feature/newhotness from the branch tab</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615" target="_blank">Unable to run multibranch pipeline with '/' the name</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-07-14T16:02:02+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>If you setup a multibranch github project based on: </p>

<p><a href="https://github.com/cloudbeers/blueocean-demo-project" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/cloudbeers/blueocean-demo-project</a></p>

<p>And try to run the branch called "feature/newhotness" you get: </p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>
jenkins-js-extension.js:10841 POST http://localhost:8080/jenkins/blue/rest/organizations/jenkins/pipelines/multi2/branches/feature%2Fnewhotness/runs/ 404 (Not Found)run @ jenkins-js-extension.js:10841run @ jenkins-js-extension.js:13760onClick @ jenkins-js-extension.js:13824ReactErrorUtils.invokeGuardedCallback @ blueocean.js:28058executeDispatch @ blueocean.js:20044executeDispatchesInOrder @ blueocean.js:20064executeDispatchesAndRelease @ blueocean.js:19517executeDispatchesAndReleaseTopLevel @ blueocean.js:19528forEachAccumulated @ blueocean.js:34385processEventQueue @ blueocean.js:19690runEventQueueInBatch @ blueocean.js:28084handleTopLevel @ blueocean.js:28095handleTopLevelImpl @ blueocean.js:28173perform @ blueocean.js:33838batchedUpdates @ blueocean.js:27265batchedUpdates @ blueocean.js:31544dispatchEvent @ blueocean.js:28250
jenkins-js-extension.js:13767 Unexpected error queuing a run of "feature%2Fnewhotness". Response:(anonymous function) @ jenkins-js-extension.js:13767(anonymous function) @ jenkins-js-extension.js:10851
jenkins-js-extension.js:13768 Response {type: "basic", url: "http://localhost:8080/jenkins/blue/rest/organizati…elines/multi2/branches/feature%2Fnewhotness/runs/", status: 404, ok: false, statusText: "Not Found"…}

</pre>
</div></div>

<p>This is possibly related to: <a href="https://issues.jenkins-ci.org/browse/JENKINS-36613" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-36613</a></p>

<p>to reproduce: </p>

<p>1) setup a multibranch github based on <a href="https://github.com/cloudbeers/blueocean-demo-project" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/cloudbeers/blueocean-demo-project</a>
2) let it run, and then try to rerun the feature/newhotness from the branch tab</p>
</pre>
</details>
</details>
<details><summary><i>2 attachments</i></summary>

- [Screen Shot 2016-07-12 at 3.40.05 PM.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/33282/Screen%20Shot%202016-07-12%20at%203.40.05%20PM.png)
  > ![Screen Shot 2016-07-12 at 3.40.05 PM.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/33282/Screen%20Shot%202016-07-12%20at%203.40.05%20PM.png)
- [Screen Shot 2016-07-12 at 3.40.24 PM.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/33281/Screen%20Shot%202016-07-12%20at%203.40.24%20PM.png)
  > ![Screen Shot 2016-07-12 at 3.40.24 PM.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/33281/Screen%20Shot%202016-07-12%20at%203.40.24%20PM.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=263824&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-263824">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> is this another one you can take a look at? </p>

<p>The encoding isn't quite right, somewhere...</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> is this another one you can take a look at? </p>

<p>The encoding isn't quite right, somewhere...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tfennelly.png" title="tfennelly's avatar" /> tfennelly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=264125&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264125">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Guys, see <a href="https://issues.jenkins.io/browse/JENKINS-36286" title="Provide generic way to get Link for given Jenkins job name" class="issue-link" data-issue-key="JENKINS-36286"><del>JENKINS-36286</del></a>. The <tt>BlueMessageEnricher</tt> has temp for adding the job URL. Looks like this can now be replaced with whatever was done for <a href="https://issues.jenkins.io/browse/JENKINS-36286" title="Provide generic way to get Link for given Jenkins job name" class="issue-link" data-issue-key="JENKINS-36286"><del>JENKINS-36286</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>Guys, see <a href="https://issues.jenkins.io/browse/JENKINS-36286" title="Provide generic way to get Link for given Jenkins job name" class="issue-link" data-issue-key="JENKINS-36286"><del>JENKINS-36286</del></a>. The <tt>BlueMessageEnricher</tt> has temp for adding the job URL. Looks like this can now be replaced with whatever was done for <a href="https://issues.jenkins.io/browse/JENKINS-36286" title="Provide generic way to get Link for given Jenkins job name" class="issue-link" data-issue-key="JENKINS-36286"><del>JENKINS-36286</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=264127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264127">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> yes, the pipeline name being sent doesn't include the folder path which results in a 404. There are also the separate encoding issues with wacky branch names.</p>

<p>Please see my comment on <a href="https://issues.jenkins.io/browse/JENKINS-36638" title="SSE events not coming through to the client" class="issue-link" data-issue-key="JENKINS-36638"><del>JENKINS-36638</del></a> as far as the fix. I started some work in the UI layer but I can't really make progress until we adjust the SSE event payload to have the proper URL.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> yes, the pipeline name being sent doesn't include the folder path which results in a 404. There are also the separate encoding issues with wacky branch names.</p>

<p>Please see my comment on <a href="https://issues.jenkins.io/browse/JENKINS-36638" title="SSE events not coming through to the client" class="issue-link" data-issue-key="JENKINS-36638"><del>JENKINS-36638</del></a> as far as the fix. I started some work in the UI layer but I can't really make progress until we adjust the SSE event payload to have the proper URL.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=264128&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264128">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly" class="user-hover" rel="tfennelly">tfennelly</a> thanks. I talked to Vivek and am going to make this change locally for now to try to unblock myself. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly" class="user-hover" rel="tfennelly">tfennelly</a> thanks. I talked to Vivek and am going to make this change locally for now to try to unblock myself. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=264188&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264188">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>May be fixed by: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/346" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/346</a> (but in a probably clumsy way)</p>
</pre>
</details>

</ul>
</details>

<p>May be fixed by: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/346" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/346</a> (but in a probably clumsy way)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=264222&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264222">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I believe the run bugs have indeed been fixed by <a href="https://issues.jenkins.io/browse/JENKINS-36675" title="REGRESSION: unable to launch any multibranch jobs" class="issue-link" data-issue-key="JENKINS-36675"><del>JENKINS-36675</del></a> (see my comment there). I'm going to create a new ticket that describes some changes to the related code I'd like to make using HAL that I hope will make the code less brittle. And I'm going to resolve this one as fixed.</p>
</pre>
</details>

</ul>
</details>

<p>I believe the run bugs have indeed been fixed by <a href="https://issues.jenkins.io/browse/JENKINS-36675" title="REGRESSION: unable to launch any multibranch jobs" class="issue-link" data-issue-key="JENKINS-36675"><del>JENKINS-36675</del></a> (see my comment there). I'm going to create a new ticket that describes some changes to the related code I'd like to make using HAL that I hope will make the code less brittle. And I'm going to resolve this one as fixed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=270169&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-270169">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Thorsten Scherler
Path:
 src/main/js/page_objects/blueocean/bluePipelineActivity.js
 src/test/js/smoke.js
<a href="http://jenkins-ci.org/commit/blueocean-acceptance-test/ab6c94fc62a8d3ec7832124fff6166d196c1151d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/blueocean-acceptance-test/ab6c94fc62a8d3ec7832124fff6166d196c1151d</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a> Add test for clicking on run button then on the emerging toast the open</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Thorsten Scherler<br/>
Path:<br/>
 src/main/js/page_objects/blueocean/bluePipelineActivity.js<br/>
 src/test/js/smoke.js<br/>
<a href="http://jenkins-ci.org/commit/blueocean-acceptance-test/ab6c94fc62a8d3ec7832124fff6166d196c1151d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/blueocean-acceptance-test/ab6c94fc62a8d3ec7832124fff6166d196c1151d</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a> Add test for clicking on run button then on the emerging toast the open</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36615?focusedId=270260&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-270260">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Thorsten Scherler
Path:
 src/main/js/page_objects/blueocean/bluePipelineActivity.js
 src/test/js/smoke.js
<a href="http://jenkins-ci.org/commit/blueocean-acceptance-test/8d50b41ab0288bf38257d471d6ce6e9fb1d89983" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/blueocean-acceptance-test/8d50b41ab0288bf38257d471d6ce6e9fb1d89983</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a> Add test for clicking on run button then on the emerging toast the open (#41)</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Thorsten Scherler<br/>
Path:<br/>
 src/main/js/page_objects/blueocean/bluePipelineActivity.js<br/>
 src/test/js/smoke.js<br/>
<a href="http://jenkins-ci.org/commit/blueocean-acceptance-test/8d50b41ab0288bf38257d471d6ce6e9fb1d89983" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/blueocean-acceptance-test/8d50b41ab0288bf38257d471d6ce6e9fb1d89983</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a> Add test for clicking on run button then on the emerging toast the open (#41)</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-35742%22">JENKINS-35742</a>]</i>
